fleet-be/src/main/java/com/everis/training/fleet/business/fleet/control/VehicleController.java

```
@@ -22,27 +23,45 @@
         return repo.getAllFreeVehicles();
     }
 
-    public void add(Vehicle vehicle) {
-        if (vehicle.getVin().length()==17){
-            repo.saveVehicle(vehicle);
+    public void delete(String vin) throws Exception{
+        try {
+            vinCheck(vin);
+            repo.deleteVehicle(vin);
+        } catch (Exception e) {
+            e.printStackTrace();
         }
     }
 
-    public void delete(String vin) {
-        if (vin.length() == 17){
-            repo.deleteVehicle(vin);
+    public void reserveCar(Integer id, String vin) throws Exception{
+        try {
+            vinCheck(vin);
+            repo.reserveCar(id, vin);
+        } catch (Exception e) {
+            e.printStackTrace();
```

You are swallowing the exception here. The exception should be thrown.